### PR TITLE
[feat (issue #6389)] make tauri icon support SVGs

### PR DESCRIPTION
**Pull request by fetzsav**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x ] No

### Checklist
- [x ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ x] I have added a convincing reason for adding this feature, if necessary

### Other information

This allows .SVG files to be passed into the CLI's icon module. This was a request in the issues tab

### CODE REVIEW

**Comment by lorenzolewis** [comment]
This is an awesome addition! @amrbashir maybe this is something we could use downstream in CTA later as well to reduce the size even more? 

Edit: Would require a few more changes to the actual core bits of Tauri that pull in the rendered icons I realise, but still cool to move closer to that possibility!

**Comment by amrbashir** [comment]
Cool! If this ends up getting merged, and tauri changes its logic to depend on a single svg icon and generate the others on the fly, it will save us a couple of hunderd kilobytes

**Comment by fetzsav** [comment]
As per discussion in the Discord, there are a couple points worth checking before merge.

1) Resizing of the SVG. Currently, we are not utilizing the SVG, as we are converting it to rasterized and then performing the image resize. I’ll look into this when I get home today.

2) Extra C dependency from nsvg package.  Per Discord discussion, resvg was suggested. Just let me know if you guys would prefer to use a different crate.

In response to the previous comments… are you guys talking about saving extra disk space? So, if an SVG is used, the PNGs arent saved to disk? Just some clarification here would help thanks!

**Comment by lorenzolewis** [comment]
@fetzsav we were just having a slight sidebar for CTA (create-tauri-app) for a few ideas of how this could help get us closer to shrinking the templates down there in the future. You'd still need to generate the raster images to use in icons, this PR just helps us get closer to using SVGs to generate those raster images and then in theory in the future we could use that functionality to optimise CTA.

tl;dr no impact to this, just some enhancements we could get closer to downstream with this PR

**Comment by lucasfernog** [comment]
I think this feature should target the v2 release, v1.3 is pretty much ready and the idea is to focus 100% on v2 when that release is out.

**Comment by lucasfernog** [comment]
I tried to use resvg but couldn't get the resize to work. Can you review @FabianLars ?

**Comment by FabianLars** [comment]
I switched back to resvg because nsvg failed to resize my very first test icon 😂 (it worked with other test files, resvg could handle all i tested). Resizing should work, but i know what issue you were talking about.

Also the ios bg color is now applied after resizing to make sure it's the svg renderer resizing it and not image-rs. Seemed to work on my end but i have no ios setup to test this rn.

**Review by FabianLars — DISMISSED** [review]

**Review by lucasfernog — APPROVED** [review]

File: tooling/cli/src/icon.rs [hunk 1]
```diff
@@ -464,14 +464,14 @@ fn resize_and_save_png(
   source: &Source,
   size: u32,
   file_path: &Path,
-  ios_color: Option<Rgba<u8>>,
+  bg_color: Option<Rgba<u8>>,
```
Comment (FabianLars) [inline tooling/cli/src/icon.rs:467]: i swear i named it this way first but reverted it for no reason 😂
